keras/layers/core.py

```diff
@@ -298,6 +300,10 @@ def output_shape(self):
             return tuple(output_shape)
         elif self.mode == 'join':
             return None 
+        elif self.mode in ['dot','cos']:
```

Comment:
The way i had implemented the shape is not elegant at all :D try something like that instead             

```python
 elif self.mode =='dot':
       return tuple(input_shapes[0][0], 1)
```